catt/stream_info.py

```diff
@@ -31,14 +31,19 @@ class StreamInfoError(Exception):
 
 
 class StreamInfo:
-    def __init__(self, video_url, model=None, device_type=None):
+    def __init__(self, video_url, model=None, device_type=None, ytdl_option=None):
         if "://" not in video_url:
             self._local_file = video_url
             self.local_ip = self._get_local_ip()
             self.port = random.randrange(45000, 47000)
             self.is_local_file = True
         else:
-            self._ydl = youtube_dl.YoutubeDL({"quiet": True, "no_warnings": True})
+            self.ytdl_params = dict()
+            if ytdl_option:
+                self.ytdl_params = dict(ytdl_option)
+            self.ytdl_params.update({"quiet": True, "no_warnings": True})
+
+            self._ydl = youtube_dl.YoutubeDL(self.ytdl_params)
```

Comment:
When the user specifies credentials for ```youtube-dl``` to use, it does not fail if those credentials are wrong/inadequate, so the current options are not really desirable in this case (I'm sure this is also the case for other options). I would prefer something like this:
```
DEFAULT_YTDL_OPTS = {"quiet": True, "no_warnings": True}
self._ydl = youtube_dl.YoutubeDL(dict(ytdl_option) if ytdl_option else DEFAULT_YTDL_OPTS)
```

Reply:
+1

Reply:
But with this approach the default parameters will be replaced unless they're explicitly written when the user pass the other ytdl parameters. So how about something like this:
```python
    ytdl_params = {"quiet": True, "no_warnings": True}
            if ytdl_options:
                ytdl_params.update(ytdl_options)
            self._ydl = youtube_dl.YoutubeDL(ytdl_params)
```

Reply:
> But with this approach the default parameters will be replaced unless they're explicitly passed with the other ytdl parameters

That is intentional. Like I said, ```{"quiet": True, "no_warnings": True}``` is not desirable when used with credentials and probably many other options.

Reply:
These options increase verbosity level, that's all. You can get any error messages with or without these parameters. Especially without the `quiet` option, the use gets any stdout of ytdl prints.

Reply:
Eh? When ```--no-warnings``` are set, ```WARNING: Unable to login: Invalid password``` is not printed.

Reply:
Youtube raises an error message when it's unable to provide the stream link. So it does not do anything with warning messages. I think there's no need for warning messages unless we want more verbosity for debugging.

The error messages I get with default options;
`ERROR: Unable to login, incorrect username and/or password`

`ERROR: Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:841)> (caused by URLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:841)'),))
Error: Remote resource not found.`

Reply:
No, I have just passed  wrong username and password

Reply:
What url did you use?

Reply:
Well, I got SSL failure dailymotion, vk and youtube. For login error, I've used vk.com

Reply:
OK, I'll have a look, after you've pushed the other changes.
(I still stand by my request though, as many options could alter the behaviour of ```catt```, and thus the extra verbosity will be useful to users.)

Reply:
@xybydy What is your setup? I cannot reproduce the failing behaviour. I only get warnings (using ```youtube-dl 2018.8.4```).

Reply:
I have no idea what's going on. @skorokithakis can you test this PR?

Reply:
Got it recorded at https://asciinema.org/a/5MDij53wAuCe709Wq7COJM5At

Reply:
Thanks, I can reproduce the error with your vk.com link. But with youtube it's only warnings.

Reply:
..using ```https://www.youtube.com/feed/subscriptions```. It might be different with personal videos.

Reply:
As I've mentioned, it seems youtube-dl raises an error when ytdl is unable to provide a stream link, not 100% sure about that though. It seems, if the wrong credentials have nothing to do with providing the stream link, only warnings show up.

Reply:
I still prefer ```dict(ytdl_option) if ytdl_option else DEFAULT_YTDL_OPTS```, as the extra verbosity is bound to be useful with a lot of these options.

Reply:
I agree with @theychx, as it will be surprising behaviour for the user if they specify options X and Y and we're secretly setting options A and B behind their back. When the user hasn't specified a preference, we can do whatever we want, but when they say they want a specific set of options, we shouldn't set anything they didn't explicitly request.